src/modules/uORB/uORBDeviceNode.cpp

```diff
@@ -378,7 +376,7 @@ int uORB::DeviceNode::unadvertise(orb_advert_t handle)
 	 * of subscribers and publishers. But we also do not have a leak since future
 	 * publishers reuse the same DeviceNode object.
 	 */
-	devnode->_advertised = false;
+	devnode->remove_publisher();
```

Comment:
After this we should be in a position where we can considering deleting the DeviceNode entirely. I believe the `_subscriber_count` is correct, although it really needs testing.

https://github.com/PX4/PX4-Autopilot/blob/ecb462f325aa4389769dcd7844b3d0602c1912be/src/modules/uORB/uORBDeviceNode.hpp#L255